depmapomics/ccle.py

```diff
@@ -0,0 +1,1452 @@
+
+    pool = multiprocessing.Pool(cores)
+    res_wes = pool.map(mapBedWES, os.listdir(wesdir))
+    sorted_guides_bed_wes = guides_bed.sort_values(
+        by=["chrom", "start", "end"]
+    ).reset_index(drop=True)
+    print("wes: done pooling")
+    for name, val in res_wes:
+        if val is not None:
+            sorted_guides_bed_wes[name] = val
+    print("saving wes matrix")
+    sorted_guides_bed_wes.to_csv(savedir + "binary_mutguides_wes.tsv.gz")
+
+    print("generating germline matrix for wgs")
+    wm = dm.WorkspaceManager(wgsrefworkspace)
+    samp = wm.get_samples()
+    vcfs = samp[vcf_colname]
+    vcfslist = vcfs[~vcfs.isna()].tolist()
+    # load vcfs using dalmatian
+    h.createFoldersFor(wgsdir)
+    cmd = [
+        "gsutil cp "
+        + sam
+        + " "
+        + wgsdir
+        + sam.split("/")[-1]
+        + "&& \
+ bcftools index "
+        + wgsdir
+        + sam.split("/")[-1]
+        + " && \
+ bcftools query \
+  --exclude \"FILTER!='PASS'&GT!='mis'&GT!~'\.'\" \
+  --regions-file "
+        + bed_location
+        + " \
+  --format '%CHROM\\t%POS\\t%END\\t%ALT{0}\n' "
+        + wgsdir
+        + sam.split("/")[-1]
+        + " >\
+ "
+        + wgsdir
+        + "loc_"
+        + sam.split("/")[-1].split(".")[0]
+        + ".bed &&\
+ rm "
+        + wgsdir
+        + sam.split("/")[-1]
+        + "*"
+        for sam in vcfslist
+    ]
+
+    h.parrun(cmd, cores=cores)
+
+    pool = multiprocessing.Pool(cores)
+    res_wgs = pool.map(mapBedWGS, os.listdir(wgsdir))
+    sorted_guides_bed_wgs = guides_bed.sort_values(
+        by=["chrom", "start", "end"]
+    ).reset_index(drop=True)
+    print("wgs: done pooling")
+    for name, val in res_wgs:
+        if val is not None:
+            sorted_guides_bed_wgs[name] = val
+    print("saving wgs matrix")
+    sorted_guides_bed_wgs.to_csv(savedir + "binary_mutguides_wgs.tsv.gz")
+
+    # merging wes and wgs
+    print("merging wes and wgs matrices")
+    wgs_mat_noguides = sorted_guides_bed_wgs.iloc[:, 4:]
+    wes_mat_noguides = sorted_guides_bed_wes.iloc[:, 4:]
+
+    pr_table = trackerobj.read_pr_table()
+    renaming_dict = dict(list(zip(pr_table.CDSID, pr_table.index)))
+
+    wgs_whitelist = [x for x in wgs_mat_noguides.columns if x in renaming_dict]
+    wes_whitelist = [x for x in wes_mat_noguides.columns if x in renaming_dict]
+    wgs_whitelist_mat = wgs_mat_noguides[wgs_whitelist]
+    wes_whitelist_mat = wes_mat_noguides[wes_whitelist]
+    wgs_whitelist_mat = wgs_whitelist_mat.rename(columns=renaming_dict)
+    wes_whitelist_mat = wes_whitelist_mat.rename(columns=renaming_dict)
+
+    mergedmat = wgs_whitelist_mat.join(wes_whitelist_mat)
+    mergedmat = mergedmat.astype(bool).astype(int)
+    sorted_mat = sorted_guides_bed_wgs.iloc[:, :4].join(mergedmat)
+    sorted_mat["end"] = sorted_mat["end"].astype(int)
+    sorted_mat.to_csv(savedir + "merged_binary_germline.csv", index=False)
+
+    from taigapy import TaigaClient
```

Comment:
make sure that imports to taiga are hidden like this (and not on top of any files) in the depmap_omics project